feat: add C implementation for `math/base/special/roundsd` (#4438)
stdlib-js/stdlib: lib/node_modules/@stdlib/math/base/special/roundsd/README.md

# roundsd

> > Rounds a double-precision floating-point number to the nearest value with `n` significant figures.

Reviewer: ```suggestion
> Round a double-precision floating-point number to the nearest value with `n` significant figures.
```

<!-- /.examples -->

<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->

<section class="related">
<!-- C interface documentation. -->

* * *

## See Also
<section class="c">

## C APIs

> -   <span class="package-name">[`@stdlib/math/base/special/ceilsd`][@stdlib/math/base/special/ceilsd]</span><span class="delimiter">: </span><span class="description">round a numeric value to the nearest number toward positive infinity with N significant figures.</span>

Reviewer: While this will get reinserted through automation, in general, for existing packages, we shouldn't be manually removing the list of related packages. There is not a reason for doing so.

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/math/base/special/roundsd.h"
```

#### stdlib_base_roundsd( x, n, b )

> Rounds a `numeric` value to the nearest `number` with `n` significant figures.

Reviewer: ```suggestion
Rounds a double-precision floating-point number to the nearest value with `n` significant figures.
```

Reviewer: It's likely that we should update the description elsewhere throughout this package.

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/math/base/special/roundsd.h"
```

#### stdlib_base_roundsd( x, n, b )

> Rounds a numeric value to the nearest number with \\(n\\) significant figures.

Reviewer 2: ```suggestion
Rounds a `numeric` value to the nearest `number` with `n` significant figures.
```

Same description as above.

```c
 double v = roundsd( 3.141592653589793, 3, 10 );

>  // returns 3.14

Reviewer 2: ```suggestion
double v = stdlib_base_roundsd( 3.141592653589793, 3, 10 );
// returns 3.14
```

```

The function accepts the following arguments:

-   **x**: `[in] double` input value.
-   **n**: `[in] int32_t` number of significant figures.
-   **b**: `[in] int32_t` base.

```c
double stdlib_base_roundsd( const double x, const int32_t n, const int32_t b );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/math/base/special/roundsd..h"
#include <stdio.h>

int main( void ) {
    const double x[] = { 3.14, -3.14, 0.0, 0.0/0.0 };

    double y;
    int i;
    for ( i = 0; i < 4; i++ ) {

>         y = stdlib_base_roundsd( x[ i ], -2 );

Reviewer 2: ```suggestion
        y = stdlib_base_roundsd( x[ i ], -2, 10 );
```
